### DISCUSSION TITLE
Is a dedicated event persistence solution the right avenue?

### DISCUSSION THREAD

**Discussion by gdupontf** [spring-projects/spring-modulith#394] [Q&A] [answered]
From what I'm gathering, there's no functionality in the event persistence projects that can't be covered by the more mature Spring Integration APIs. Am I wrong? If anything, the spring-messaging API by itself can solve some inherent problems like #258 and spring-integration takes care of a lot of plumbing.

**Comment by odrotbohm** [comment, marked as answer]
None of the other Spring projects provide out of the box support for reliable async, transactional, in-JVM event publication. By definition, the focus of Spring Integration is *integrating* individual applications, while Spring Modulith is focused on structuring those individual applications internally.
